packages/cli/commands/project/logs.js

```diff
@@ -4,253 +4,154 @@ const {
 // const { outputLogs } = require('../../lib/ui/serverlessFunctionLogs');
-const {
-  fetchProject,
-  fetchDeployComponentsMetadata,
-} = require('@hubspot/local-dev-lib/api/projects');
 const { getTableContents, getTableHeader } = require('../../lib/ui/table');
-// const {
-//   logApiErrorInstance,
-//   ApiErrorContext,
-// } = require('../../lib/errorHandlers/apiErrors');
-// const {
-//   getFunctionLogs,
-//   getLatestFunctionLog,
-//   getProjectAppFunctionLogs,
-//   getLatestProjectAppFunctionLog,
-// } = require('@hubspot/local-dev-lib/api/functions');
+const { logApiErrorInstance } = require('../../lib/errorHandlers/apiErrors');
 
-const { ensureProjectExists } = require('../../lib/projects');
 const { loadAndValidateOptions } = require('../../lib/validation');
 const { uiBetaTag, uiLine, uiLink } = require('../../lib/ui');
 const { projectLogsPrompt } = require('../../lib/prompts/projectsLogsPrompt');
 // const { tailLogs } = require('../../lib/serverlessLogs');
 const { i18n } = require('../../lib/lang');
-// const { EXIT_CODES } = require('../../lib/enums/exitCodes');
+const { EXIT_CODES } = require('../../lib/enums/exitCodes');
+const ProjectLogsManager = require('../../lib/projectLogsManager');
 
 const i18nKey = 'commands.project.subcommands.logs';
 
 const getPrivateAppsUrl = accountId => {
   const baseUrl = getHubSpotWebsiteOrigin(
-    getEnv() === 'qa' ? ENVIRONMENTS.QA : ENVIRONMENTS.PROD
+    getEnv(accountId) === 'qa' ? ENVIRONMENTS.QA : ENVIRONMENTS.PROD
   );
 
   return `${baseUrl}/private-apps/${accountId}`;
 };
 
-// We currently cannot fetch logs directly to the CLI. See internal CLI issue #413 for more information.
-
-// const handleLogsError = (e, name, projectName) => {
-//   if (e.response && e.response.status === 404) {
-//     logger.debug(`Log fetch error: ${e.message}`);
-//     logger.log(i18n(`${i18nKey}.logs.noLogsFound`, { name }));
-//   } else {
-//     logApiErrorInstance(
-//       e,
-//       new ApiErrorContext({ accountId: getAccountId(), projectName })
-//     );
-//     process.exit(EXIT_CODES.ERROR);
-//   }
-// };
-
-// const handleFunctionLog = async (accountId, options) => {
-//   const {
-//     latest,
-//     follow,
-//     compact,
-//     appPath,
-//     functionName,
-//     projectName,
-//   } = options;
-
-//   let logsResp;
+const handleFunctionLog = async (accountId, options) => {
+  const { functionName } = options;
 
-//   const tailCall = async after => {
-//     try {
-//       return appPath
-//         ? getProjectAppFunctionLogs(
-//             accountId,
-//             functionName,
-//             projectName,
-//             appPath,
-//             {
-//               after,
-//             }
-//           )
-//         : getFunctionLogs(accountId, functionName, { after });
-//     } catch (e) {
-//       handleLogsError(e, functionName, projectName);
-//     }
-//   };
+  // const { latest, follow, compact, functionName } = options;
```

Comment:
If you want, you can totally just delete all this commented code. We have the commit history if we ever want to go back and reference it

Reply:
https://github.com/HubSpot/hubspot-cli/pull/1131/commits/f88732c070b9f991f8fbdd5ee8094ec353328126